hermes/bin/src/runtime/host/hermes/hash/blake2b.rs

```diff
@@ -0,0 +1,224 @@
+//! Implementation of blake2b and blake2bmac hash functions.
+
+use blake2b_simd::Params;
+
+use crate::runtime::extensions::bindings::hermes::{binary::api::Bstr, hash::api::Errno};
+
+/// Implementation of blake2b.
+///
+/// **Parameters**
+///
+/// - `buf`: Bytes string
+/// - `outlen`: Optional output length, if not specified it is set to 64.
+///
+/// **Returns**
+///
+/// - `ok(Bstr)`: Hash of bytes string.
+/// - `error(Errno)`:
+///     - `InvalidDigestByteLength` if an `outlen` is 0.
+///     - `HashTooBig` if `outlen` is greater than 64.
+pub(crate) fn blake2b_impl(buf: &Bstr, outlen: Option<u8>) -> Result<Bstr, Errno> {
+    // Default to 64 bytes Blake2b-512
+    let outlen: usize = outlen.unwrap_or(64).into();
+
+    // outlen is set, but invalid when == 0
+    if outlen == 0 {
+        return Err(Errno::InvalidDigestByteLength);
+    } else if outlen > 64 {
+        return Err(Errno::HashTooBig);
+    }
+    let hash = Params::new()
+        .hash_length(outlen)
+        .to_state()
+        .update(buf)
+        .finalize();
+
+    return Ok(hash.as_bytes().into());
+}
+
+/// Implementation of blake2bmac given a buffer, outlen, key, salt, and persona.
+///
+/// **Parameters**
```

Comment:
Update the format to the following:
```
# Arguments
- ...

# Errors
- ...
```
It is a common way for Rust how to describe such sections https://doc.rust-lang.org/book/ch14-02-publishing-to-crates-io.html#commonly-used-sections.